service/src/main/java/uk/gov/hmcts/reform/fpl/service/GatekeepingOrderService.java

```diff
@@ -0,0 +1,117 @@
+import uk.gov.hmcts.reform.fpl.model.SaveOrSendGatekeepingOrder;
+import uk.gov.hmcts.reform.fpl.model.StandardDirectionOrder;
+import uk.gov.hmcts.reform.fpl.model.common.JudgeAndLegalAdvisor;
+import uk.gov.hmcts.reform.fpl.model.event.GatekeepingOrderEventData;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.commons.lang3.ObjectUtils.defaultIfNull;
+import static org.apache.commons.lang3.ObjectUtils.isEmpty;
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import static uk.gov.hmcts.reform.fpl.enums.OrderStatus.DRAFT;
+import static uk.gov.hmcts.reform.fpl.model.common.DocumentReference.buildFromDocument;
+import static uk.gov.hmcts.reform.fpl.utils.JudgeAndLegalAdvisorHelper.buildAllocatedJudgeLabel;
+import static uk.gov.hmcts.reform.fpl.utils.JudgeAndLegalAdvisorHelper.getJudgeForTabView;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class GatekeepingOrderService {
+
+    public SaveOrSendGatekeepingOrder buildSaveOrSendPage(CaseData caseData, Document document) {
+        //add draft document
+        SaveOrSendGatekeepingOrder saveOrSendGatekeepingOrder = caseData.getGatekeepingOrderEventData()
+            .getSaveOrSendGatekeepingOrder().toBuilder()
+            .draftDocument(buildFromDocument(document))
+            .orderStatus(null)
+            .build();
+
+        saveOrSendGatekeepingOrder = saveOrSendGatekeepingOrder.toBuilder()
+            .nextSteps(buildNextStepsLabel(caseData))
+            .build();
+
+        return saveOrSendGatekeepingOrder;
+    }
+
+    public JudgeAndLegalAdvisor setAllocatedJudgeLabel(Judge allocatedJudge, JudgeAndLegalAdvisor issuingJudge) {
+        String assignedJudgeLabel = buildAllocatedJudgeLabel(allocatedJudge);
+
+        return issuingJudge.toBuilder().allocatedJudgeLabel(assignedJudgeLabel).build();
+    }
+
+    private boolean hasEnteredIssuingJudge(JudgeAndLegalAdvisor judgeAndLegalAdvisor) {
+        if (isEmpty(judgeAndLegalAdvisor)) {
+            return false;
+        }
+
+        if (judgeAndLegalAdvisor.isUsingAllocatedJudge()) {
+            return true;
+        } else {
+            //after judge title selected the other fields are mandatory, so checking title verifies judge entry (?)
+            return isNotEmpty(judgeAndLegalAdvisor.getJudgeTitle());
+        }
+    }
+
+    //this constructs a label which hides the option to seal if mandatory information is missing
+    //previous button can break this functionality as logic uses a hidden field (EUI-3922)
+    private String buildNextStepsLabel(CaseData caseData) {
+        List<String> requiredMissingInformation = new ArrayList<>();
+
+        if (caseData.getFirstHearing().isEmpty()) {
+            requiredMissingInformation.add("* the first hearing details");
+        }
+
+        if (isEmpty(caseData.getAllocatedJudge())) {
+            requiredMissingInformation.add("* the allocated judge");
+        }
+
```

Comment:
maybe allocation decision is also mandatory (but this is basically usurped by allocated judge so...)

Reply:
can worry about this later if it's required (when adding the allocation decision page)